lib/THC/THCTensorMath2.cu

```diff
@@ -660,3 +660,47 @@ void THCudaTensor_randn(THCState *state, THCudaTensor *r_, THLongStorage *size)
   THCudaTensor_resize(state, r_, size, NULL);
   THCudaTensor_normal(state, r_, 0, 1);
 }
+
+struct TensorCrossOp {
+  TensorCrossOp(long sx, long sy, long so) : sx(sx), sy(sy), so(so) {}
+
+  __device__ __forceinline__ void operator()(float* out, float* x, float*y) {
+    *(out+0*so) = *(x+1*sx) * (*(y+2*sy)) - *(x+2*sx) * (*(y+1*sy));
+    *(out+1*so) = *(x+2*sx) * (*(y+0*sy)) - *(x+0*sx) * (*(y+2*sy));
```

Comment:
a nit:

```out[1 * so] = x[2 * sx] * y[0 * sy] - x[0 * sx] * y[2 * sy];```

etc. looks much more readable to me.

also, please float the binary operators in whitespace?